DB/src/main/java/io/deephaven/db/v2/remote/ConstructSnapshot.java

```diff
@@ -1241,18 +1282,19 @@ public static boolean serializeAllTable(final boolean usePrev,
                 }
 
                 final BarrageMessage.AddColumnData acd = new BarrageMessage.AddColumnData();
-                snapshot.addColumnData[jj] = acd;
+                snapshot.addColumnData[ii] = acd;
 
                 final ColumnSource<?> columnSource = table.getColumnSource(columnSources[ii]);
                 acd.data = getSnapshotDataAsChunk(columnSource, sharedContext, snapshot.rowsIncluded, usePrev);
                 acd.type = columnSource.getType();
+                acd.componentType = columnSource.getComponentType();
             }
         }
 
         log.info().append(System.identityHashCode(logIdentityObject))
                 .append(": Snapshot candidate step=").append((usePrev ? -1 : 0) + LogicalClock.getStep(getConcurrentAttemptClockValue()))
                 .append(", rows=").append(snapshot.rowsIncluded).append("/").append(positionsToSnapshot)
-                .append(", cols=").append(FormatBitSet.formatBitSet(snapshot.addColumns))
+                .append(", cols=").append(FormatBitSet.formatBitSet(columnsToSerialize))
```

Comment:
This doesn't like `null` we don't think. So maybe replace with `columnsToSerialize == null ? "ALL" : FormatBitSet.formatBitSet(columnsToSerialize)`